### PR TITLE
Fix the wheel build

### PR DESCRIPTION
**Pull request by kiyo-masui**

### CODE REVIEW

**Comment by kiyo-masui** [comment]
@ketiltrout Just trying to fix the release system. I solved a few problems, but the wheel build still fails when it's tested... looks like the tests can't import the modules.

See: https://github.com/kiyo-masui/bitshuffle/actions/runs/11016468354

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.12"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```
Comment (DimitriPapadopoulos) [inline .github/workflows/main.yml:15]: At this point, you might as well add `"3.13"`. Pre-release versions are already available in GitHub runners.

**Comment by DimitriPapadopoulos** [comment]
See also #142.

**Comment by ljgray** [comment]
Ok, this should be fixed. See https://github.com/kiyo-masui/bitshuffle/actions/runs/11022245770.

The underlying issue with the build is that it was trying to install hdf5 version 1.14.4, but there actually aren't any distributions available - we have to include the patch number, the most recent of which is `-3`. As a result, hdf5 wasn't getting installed and bitshuffle was failing when trying to import it. In order to fix this, I:
- specified hdf5 version 1.14.4-3 in the workflow
- modified `install_hdf5.sh` to handle a patch version

I also had to add a minimum `python_requires` to `setup.py`. `cibuildwheel` uses this to decide which version to build on, and in the absence of any specification it seemed to be falling back to python 3.6.

Finally, I updated the build requirements to support numpy 2.0. According to [numpy 2.0 dependency docs](https://numpy.org/doc/stable//dev/depending_on_numpy.html#numpy-2-0-specific-advice), wheels built against numpy 2.x will be compatible with numpy 1.xx. numpy 2.0.0rc1 is the first numpy 2 release candidate, so building against that (at minimum) should still support numpy 1.xx. I also updated the minimum supported numpy to 1.24, based on the [numpy drop schedule](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule).

**Comment by kiyo-masui** [comment]
Wow - thank you! I struggled with this for a few hours but no way I would have figured this out!

**Pull request merged**